homeassistant/components/switch/verisure.py

```diff
@@ -0,0 +1,67 @@
+"""
+homeassistant.components.switch.verisure
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Support for Verisure Smartplugs
+
+Configuration:
+
+switch:
+  platform: verisure
+
+Variables:
+
+"""
+import logging
+
+import homeassistant.components.verisure as verisure
+from homeassistant.components.switch import SwitchDevice
+
+DEPENDENCIES = ['verisure']
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Sets up the Arduino platform. """
+
+    if not verisure.MY_PAGES:
+        _LOGGER.error('A connection has not been made to Verisure mypages.')
+        return False
+
+    switches = []
+
+    switches.extend([
+        VerisureSmartplug(value)
+        for value in verisure.get_smartplug_status().values()
+        ])
+
+    add_devices(switches)
+
+
+class VerisureSmartplug(SwitchDevice):
+    """ Represents a Verisure smartplug. """
+    def __init__(self, smartplug_status):
+        self._id = smartplug_status.id
+
+    @property
+    def name(self):
+        """ Get the name (location) of the smartplug. """
+        return verisure.get_smartplug_status()[self._id].location
```

Comment:
Is `get_smartplug_status` memoized or will the device be queried twice when the device updates? (once for `name`, once for `is_on`)

To work around this, you can implement the `update` method to store the status.
```python
def update(self):
  self._status = verisure.get_smartplug_status()[self._id]

@property
def name(self):
  return self._status.location
```